src/library.js

```diff
@@ -3252,27 +3252,20 @@ mergeInto(LibraryManager.library, {
     assert(getWasmTableEntry(ptr), 'missing table entry in dynCall: ' + ptr);
 #endif
 #if MEMORY64
-    // With MEMORY64 we have an additional step to covert `p` arguments to
+    // With MEMORY64 we have an additional step to convert `p` arguments to
     // bigint. This is the runtime equivalent of the wrappers we create for wasm
     // exports in `emscripten.py:create_wasm64_wrappers`.
-    if (sig.includes('p')) {
-      var new_args = [];
-      args.forEach((arg, index) => {
-        if (sig[index + 1] == 'p') {
-          arg = BigInt(arg);
-        }
-        new_args.push(arg);
-      });
-      args = new_args;
+    for(let i = 0; i < sig.length; ++i) {
+      if (sig[i] == 'p') args[i] = BigInt(args[i]);
     }
 #endif
-    var rtn = getWasmTableEntry(ptr).apply(null, args);
+    let rtn = getWasmTableEntry(ptr)(...args);
```

Comment:
I don't think we are generally allowing the spread operator yet in generated code because closure can't easily lower it away on older JS runtime.    Same for `let`, we are still using `var` in most places.

See https://github.com/emscripten-core/emscripten/blob/5aee12332365b6a1a36254605edab065542da937/test/test_other.py#L12576 if you want to try to extend the ES6 things that we support.

Reply:
oh, gotcha. Reverted to remove let and ...args for now.